scripts/cloudhealth/add_billing_role.py

```diff
@@ -0,0 +1,223 @@
+    parser.add_argument('-v', action='count', dest='verbose',
+                        help='verbosity. up to -vvv supported.')
+
+    args = parser.parse_args()
+    return args
+
+def get_account_number():
+    ''' get AWS Account number from STS '''
+    session = boto3.Session(profile_name=os.environ['AWS_PROFILE'])
+    sts = session.client('sts')
+    query = sts.get_caller_identity()
+    return query['Account']
+
+def setup_role_policy(account_number, name='default'):
+    ''' create the role and policy in AWS '''
+
+    role_policy_document = load_json('%s.json' % name)
+    policy_document = load_json('%s.json' % name)
+    policy_arn = 'arn:aws:iam::%s:policy/%s' % (account_number, name)
+
+    session = boto3.Session(profile_name=os.environ['AWS_PROFILE'])
+    iam = session.client('iam')
+
+    check_role = None
+    try:
+        check_role = iam.get_role(RoleName=name)
+        LOG.info(check_role)
+    except botocore.exceptions.ClientError as exc:
+        LOG.debug(exc.message)
+
+    if not check_role:
+        # pylint: disable=line-too-long
+        resp = iam.create_role(RoleName=name,
+                               AssumeRolePolicyDocument=json.dumps(role_policy_document))
+        LOG.info(resp)
+        LOG.warn('AWS Role created.')
+    else:
+        LOG.warn('AWS Role found. Skipping creation.')
+
+    check_policy = None
+    try:
+        check_policy = iam.get_policy(PolicyArn=policy_arn)
+        LOG.info(check_policy)
+    except botocore.exceptions.ClientError as exc:
+        LOG.debug(exc.message)
+
+    if not check_policy:
+        resp = iam.create_policy(PolicyName=name,
+                                 PolicyDocument=json.dumps(policy_document))
+        LOG.info(resp)
+        LOG.warn('AWS Policy created.')
+    else:
+        LOG.warn('AWS Policy found. Skipping creation.')
+
+    check_role_policy = None
+    try:
+        check_role_policy = iam.get_role_policy(RoleName=name,
+                                                PolicyName=name)
+        LOG.info(check_role_policy)
+    except botocore.exceptions.ClientError as exc:
+        LOG.debug(exc.message)
+
+    if not check_role_policy:
+        resp = iam.attach_role_policy(RoleName=name, PolicyArn=policy_arn)
+        LOG.info(resp)
+        LOG.warn('AWS Role Policy created.')
+    else:
+        LOG.warn('AWS Role Policy found. Skipping creation.')
+
+def setup_cloudhealth_account(api_key, account_number, external_id=None):
+    ''' create an account listing in cloudhealth '''
+    cht = CloudHealthAwsAccount(api_key=api_key,
+                                owner_id=u'%s' % str(account_number))
+    LOG.info('CloudHealth Account: %s', cht)
+    # pylint: disable=line-too-long
+    if not getattr(cht, 'id', None):
+        cht.authentication = {'assume_role_external_id': external_id,
+                              'protocol': 'assume_role',
+                              'assume_role_arn': 'arn:aws:iam::%s:role/oso-billing' % \
```

Comment:
Looks like the oso-billing part is a parameter ("must match the filename") but I don't see that present here.  Should this be "aws_name"?

Reply:
Yes, it should be. Fixing.